pandas/tests/io/test_sql.py

```diff
@@ -245,6 +243,43 @@ def create_and_load_types(conn, types_data: list[dict], dialect: str):
             conn.execute(stmt)
 
 
+def create_and_load_postgres_datetz(conn):
```

Comment:
One thing I've been kind of careful in while setting up the ADBC connectors is an unecessary reliance on SQLAlchemy, since it would be nice to run the test suite without that dependency when not required.

Not a blocker for this PR just a larger consideration; with the ADBC change in particular I think I'd fall back to writing postgres-specific SQL for this

Reply:
That's a good point. I can do another pass after this PR to try to refactor places where data is loaded to just use raw SQL